protected/controllers/ModuleController.php

```diff
@@ -58,24 +58,23 @@ protected function performAjaxValidation($model)
 
     public function actionSaveLesson()
     {
-        $teacher = Yii::app()->user->getId();
-
-        $newOrder = Lecture::model()->addNewLesson(
-            $_POST['idModule'],
-            $_POST['titleUa'],
-            $_POST['titleRu'],
-            $_POST['titleEn'],
-            Teacher::model()->find('user_id=:user', array(':user' => $teacher))->teacher_id
+        $newLecture = array (
+            'titleUa' => Yii::app()->request->getParam('titleUa', ''),
+            'titleRu' => Yii::app()->request->getParam('titleRu', ''),
+            'titleEn' => Yii::app()->request->getParam('titleEn', ''),
+            'order' => Yii::app()->request->getParam('order', 1)
         );
 
-        Module::model()->updateByPk($_POST['idModule'], array('lesson_count' => $_POST['order']));
-        Yii::app()->user->setFlash('newLecture', 'Нова лекція №' . $newOrder . $_POST['titleUa'] . 'додана до цього модуля');
-        $idLecture = Lecture::model()->findByAttributes(array('idModule' => $_POST['idModule'], 'order' => $newOrder))->id;
+        //throw error if idModule is '0' or unset?
+        $idModule = Yii::app()->request->getParam('idModule');
 
-        LecturePage::addNewPage($idLecture, 1);
+        $model = Module::model()->findByPk($idModule);
 
```

Comment:
the same as above, findbyPk can return null - in this case we should raise the exception